docs/stock/data-grouping.md

```diff
@@ -1,24 +1,93 @@
+Data grouping replaces a sequence of data points in a series with one grouped point. The values of each grouped point is calculated from the original values of every point used.
+In Highstock charts, data grouping is done based on the pixel width of the horizontal axis, specifically through the `groupPixelWidth`. Essentially, each group on the chart holds a pixel width on the x-axis, The default value varies depending on the type of the series, because it is easier to fit more points in e.g. line chart compared to column. If the `groupPixelInterval` is set to 2 pixels, this means if your chart is displayed within 1000 pixels, it can hold up to 500 data points at once, organized according to the chart's range.
 
-![data-grouping.png](data-grouping.png)
+```js
+Highcharts.stockChart('container', {
+  series: [{
+    dataGrouping: {
+      groupPixelWidth: 5
+    },
+    data: [...]
+  }]
+});
+```
 
-By default, the grouping [approximation](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping.approximation) depends on the type of series:
 
+Data grouping in Highcharts Stock can be turned on or off using the enabled option. By default, it's set to true.
+
+```js
+Highcharts.stockChart('container', {
+    series: [{
+        dataGrouping: {
+            enabled: false
+        },
+        data: [...]
+    }]
+});
+```
+
+Here is an example of a difference between two series with the same data, but one of them is grouped:
+
+<iframe width="600" height="450" src="https://www.highcharts.com/samples/embed/stock/plotoptions/series-datagrouping-enabled" allowfullscreen></iframe>
+
+#### Overview of the most important data grouping options
+
+##### [Approximation](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping.approximation)
+With the `approximation` option, you can decide how the final value for each group should be calculated. By default it depends on the series type: 
 *   Line type series use an _average_ approximation.
 *   Column type series compute the _sum_.
 *   Range type series compute the union's _range_.
 *   OHLC type series compute the union's open, high, low and close (OHLC) values.
 
-Grouping is activated when there are many data points in the chart. As well as increasing performance it makes it easier to spot trends in a chart.
 
-Data grouping is a Highcharts Stock feature and is enabled by default. To see dataGrouping options see the [API reference](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping).
+but you can use the custom function to calculate the final value for each group. Here is an example of custom approximation function:
+
+<iframe width="600" height="600" src="https://www.highcharts.com/samples/embed/stock/plotoptions/series-datagrouping-approximation" allowfullscreen></iframe>
```

Comment:
Decrease the height:

![image](https://github.com/user-attachments/assets/f1a1e4bc-4f68-4076-bdb1-42dd83cb70f3)

Reply:
Did you decrease the height?